pkg/util/mount/mount_linux.go

```diff
@@ -773,11 +773,11 @@ func doBindSubPath(mounter Interface, subpath Subpath, kubeletPid int) (hostPath
 
 	// Create target of the bind mount. A directory for directories, empty file
 	// for everything else.
-	t, err := os.Lstat(subpath.Path)
+	fileType, err := mounter.GetFileType(subpath.Path)
```

Comment:
does this follow symlinks?

Reply:
looks like some implementations do:
```
outputBytes, err := mounter.ne.Exec("stat", []string{"-L", `--printf "%F"`, pathname}).CombinedOutput()
```

the purpose of this check is to avoid following symlinks. we cannot switch it to a call that follows them.

Reply:
This call is meant to query the file type of the subpath target so that we can create our bind mount target as the same time.  It's intentionally an unsafe call here.  The worst that can happen is the file type is wrong, and the bind mount will fail, or the mount will fail because the path doesn't exist.

But actually, I think there's a better way of fixing this.  We can switch the order.  Do the safe open first, and then create the bind mount target based off of the proc fd file.

@rmartinez3 how are you testing this fix? with which volume types?

Reply:
> @rmartinez3 how are you testing this fix? with which volume types?

it would be good to see accompanying tests that demonstrate the failure in our existing CI if possible

Reply:
This sounds like a basic failure, in which case, our existing e2e tests should cover it.  The problem is they're just not run against a containerized kubelet environment.

Reply:
Updated a test cluster and noticed some pods were not starting up. Which were the ones using hostpath type volumemounts with subpaths. Each pod was saying error  "Error: failed to prepare subPath for volumeMount ". After applying the fix i saw that the volume mounts was working and pods were running.

Reply:
You may have gotten lucky because your hostpath volume is using a path that also exists in the kubelet container.  As pointed out [here](https://github.com/kubernetes/kubernetes/issues/61456#issuecomment-375648214), hostpath on containerized kubelet has never fully worked.